protocols/pop3/src/main/java/org/apache/james/protocols/pop3/mailbox/MessageMetaData.java

```diff
@@ -59,4 +59,15 @@ public long getSize() {
         return size;
     }
     
+    /**
+     * Check whether POP3 UID is compatible with RFC1939
+     */
+    private static boolean isRFC1939Compatible(String uid) {
+        if (uid == null) {
+            return false;
+        }
+
+        return IntStream.range(0, uid.length())
+            .allMatch(i -> uid.charAt(i) >= 0x21 && uid.charAt(i) <= 0x7E);
```

Comment:
Guava comes with some handly `CharMatcher` with a nice allMatch method that for sure will perform better than its string based counterpart

Reply:
OK, I just moved the existing code here, but might as well clean it up.

Reply:
Yes that's a trap when moving old code: it shows up in the reviews and people complains! :-)